ci/test/docs-widgets/json-parser.test.js

```diff
@@ -0,0 +1,249 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file at the root of this repository.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+import { beforeEach, describe, it, expect } from "vitest";
+import { JSDOM } from "jsdom";
+import fs from "fs";
+import path from "path";
+
+describe("JSON Widget", () => {
+  let window;
+
+  // Setup before each test
+  beforeEach(() => {
+    const htmlPath = path.join(
+      __dirname,
+      "../../../doc/user/layouts/shortcodes/json-parser.html"
+    );
+    const htmlContent = fs.readFileSync(htmlPath, "utf-8");
+    const dom = new JSDOM(htmlContent, {
```

Comment:
This is a very manual vitest setup, you might consider adding a vitest.config.js and setting the [environment](https://vitest.dev/config/#environment) to jsdom so you don't have to set this up manually for each test file.

Reply:
I actually tried to do that initially, but for some reason could not get it to work that way. Happy to show you the setup that I had if you have the time today.

Reply:
I missed you this morning, but maybe we look at this first thing tomorrow my time? I usually start at 8:30 PDT.

Reply:
Sounds good, that would be great!